pkg/services/auth/auth.go

```diff
@@ -157,16 +157,16 @@ func (a *authSvc) setupDeployKey(ctx context.Context, name SecretName, targetNam
 		return pubKey, nil
 	}
 
-	return a.provisionDeployKey(ctx, targetName, name, repo)
+	return a.provisionDeployKey(ctx, targetName, name, repo, isInstall)
 }
 
-func (a *authSvc) provisionDeployKey(ctx context.Context, targetName string, name SecretName, repo gitproviders.RepoURL) (*ssh.PublicKeys, error) {
+func (a *authSvc) provisionDeployKey(ctx context.Context, targetName string, name SecretName, repo gitproviders.RepoURL, isInstall bool) (*ssh.PublicKeys, error) {
 	visibility, err := a.gitProvider.GetRepoVisibility(ctx, repo)
 	if err != nil {
 		return nil, fmt.Errorf("error getting repo visibility: %w", err)
 	}
 
-	if *visibility == gitprovider.RepositoryVisibilityPublic {
+	if *visibility == gitprovider.RepositoryVisibilityPublic && !isInstall {
```

Comment:
@J-Thompson12 Wouldn't removing this ` *visibility == gitprovider.RepositoryVisibilityPublic` logic give us the result we are looking for?

We need a deploy key any time we are writing to a repo (regardless of it's visibility), correct?

Reply:
We only need the deploy on install. We dont need it when we are adding a repo because we dont make changes to those repos.

Reply:
This samw code is used for add and install so i needed some way to say which one was using it. Since this is a temp fix till a decision is made i went the easy route with the variable.

Reply:
The statement that we don't need to deploy keys on app repos is temporarily valid since we'll need those when implementing automated image updates, which was in the wish list in the retro we had a couple of days ago.

So, probably removing the visibility validation seems to me as a better long-term solution.